packages/web/src/store/playlist-library/helpers.ts

```diff
@@ -196,6 +229,10 @@ export const removePlaylistLibraryDuplicates = (
   for (const item of library.contents) {
     switch (item.type) {
       case 'folder': {
+        if (ids.has(item.id)) {
```

Comment:
worth a comment here explaining the need to break if it's already in the set. I'm probably the original sinner here, but it took me a bit to remember what this jazz was about. Impressed that you've grokked it so well :)